src/main/java/seedu/exercise/logic/commands/statistic/StatsFactory.java

```diff
@@ -0,0 +1,281 @@
+        ArrayList<Exercise> filteredExercise = getFilteredExercise();
+        HashMap<String, Double> data = new HashMap<>();
+
+        for (Exercise e : filteredExercise) {
+            Name name = e.getName();
+            Unit unit = e.getUnit();
+            String nameWithUnit = name.toString() + " (" + unit.toString() + ")";
+            double count = 1;
+
+            if (data.containsKey(nameWithUnit)) {
+                count = data.get(nameWithUnit) + 1;
+            }
+
+            data.put(nameWithUnit, count);
+        }
+
+        return data;
+    }
+
+    /**
+     * Compute exercise quantity with filtered exercises list.
+     */
+    private HashMap<String, Double> getTotalExerciseQuantity() {
+        ArrayList<Exercise> filteredExercise = getFilteredExercise();
+        HashMap<String, Double> data = new HashMap<>();
+
+        for (Exercise e : filteredExercise) {
+            Name name = e.getName();
+            Unit unit = e.getUnit();
+            String nameWithUnit = name.toString() + " (" + unit.toString() + ")";
+            double quantity = Double.parseDouble(e.getQuantity().toString());
+            if (data.containsKey(nameWithUnit)) {
+                quantity = data.get(nameWithUnit) + quantity;
+            }
+
+            data.put(nameWithUnit, quantity);
+        }
+
+        return data;
+    }
+
+    /**
+     * Compute calories with filtered exercises list.
+     */
+    private HashMap<String, Double> getTotalCaloriesData() {
+        ArrayList<Exercise> filteredExercise = getFilteredExercise();
+        HashMap<String, Double> data = new HashMap<>();
+
+        for (Exercise e : filteredExercise) {
+            String name = e.getName().toString();
+            double calories = Double.parseDouble(e.getCalories().toString());
+
+            if (data.containsKey(name)) {
+                double temp = data.get(name);
+                calories += temp;
+            }
+
+            data.put(name, calories);
+        }
+
+        return data;
+    }
+
+    /**
+     * Get all exercises between start and end date.
+     */
+    private ArrayList<Exercise> getFilteredExercise() {
+        ArrayList<Exercise> filteredExercise = new ArrayList<>();
+        for (Exercise e : exercises) {
+            Date date = e.getDate();
+            if (Date.isBetweenStartAndEndDate(date, startDate, endDate)) {
+                filteredExercise.add(e);
+            }
+        }
+
+        return filteredExercise;
+    }
+
+    /**
+     * Get list of names from data computed.
+     */
+    private ArrayList<String> hashMapNameToList(HashMap<String, Double> data) {
+        return new ArrayList<>(data.keySet());
+    }
+
+    /**
+     * Get list of values from data computed.
+     */
+    private ArrayList<Double> hashMapDoubleToList(HashMap<String, Double> data, ArrayList<String> names) {
+        ArrayList<Double> values = new ArrayList<>();
+
+        for (String n : names) {
+            values.add(data.get(n));
```

Comment:
can be expressed as:
return new ArrayList<>(data.values())

Reply:
I need it to correspond to the list of strings, so i did it this way.